autoload/nvimdev.vim

```diff
@@ -145,9 +147,6 @@ function! nvimdev#init(path) abort
   augroup nvimdev
     autocmd!
     autocmd BufRead,BufNewFile *.h set filetype=c
-    if get(g:, 'nvimdev_auto_lint', 1)
```

Comment:
I have neomake auto open qf/location list hence when I run Neomake! make, I see linting failures on top of make errors (hence https://github.com/neomake/neomake/issues/1043) and overall as I start developing (messy progress) I find linting distracting.
I find the option name 'nvimdev_auto_lint' misleasing since the linter maker is enabled whatever the value, the current behavior matches more 'nvimdev_autorun_neomake' ?!

I've updated the patch to show what I mean though IMO we could get rid of
```
    if get(g:, 'nvimdev_autorun_neomake', 0)
      autocmd BufWritePost *.c,*.h,*.vim Neomake
    endif
```
as Neomake users would most likely have it.